### DISCUSSION TITLE
Telemetry: Whats it for and why is it included?

### DISCUSSION THREAD

**Discussion by Jamesking56** [duncanmcclean/simple-commerce#1173] [Help]
Hello,

Firstly, nice addon.

I noticed that you have some code in here which "phones home" called Telemetry which sends the total amount of money taken using your addon as well as some other production version information.

What is your intention with this data and why is this not been made transparent via your readme file?

Cheers.

**Comment by duncanmcclean** [comment]
The telemetry payload doesn't contain any personal information and even the app URL is hashed before it reaches the server (it's just so there's some kind of unique key to prevent duplicate requests).

The telemetry exists in order to provide usage data, like how many sites are actively using older versions of the addon, how much £ is being processed across all SC sites, so I can try and paint a picture about what types of sites are using the product (is it just really small shops, or medium sized ones as well).

Statamic itself also has a level of telemetry which gets sent (and can't be disabled), which is used for version info/licensing information.

It's enabled by default because everything is anonomized. I have no idea which sites in the database are which sites in production. In fact, here's a screenshot from the database:

![CleanShot 2024-11-14 at 10 43 04](https://github.com/user-attachments/assets/d9402ba0-a607-4e0e-899a-8d6d42e1d8a5)

You're very welcome to [disable it](https://simple-commerce.duncanmcclean.com/telemetry#content-opt-out-of-telemetry) if you want.

**Reply by Jamesking56** [reply]
Maybe this needs a mention in your Readme.md then and the `true` setting adding to config. I didn't know you had Telemetry in it until I noticed the Telemetry PHP class in the source.

I'm fine with you doing this (and I think most people will since it is anonymized) but its just sketchy that its not mentioned in the main readme or made clear that you are doing this by default.

I hope that makes sense

**Reply by duncanmcclean** [reply]
I'll have a think about the best place to put it, thanks!
